chainer/training/extensions/print_report.py

```diff
@@ -24,10 +24,19 @@ class PrintReport(extension.Extension):
     """
 
     def __init__(self, entries, log_report='LogReport', out=sys.stdout):
+        if not hasattr(out, "write"):
+            raise TypeError(
+                "Output stream should support `write' method!")
+
         self._entries = entries
         self._log_report = log_report
         self._out = out
 
+        if hasattr(out, "flush"):
+            self._flush = lambda: out.flush()
+        else:
+            self._flush = lambda: None
```

Comment:
You don't have to make an attribute here.

Reply:
fixed.